csrc/scheduler/matmul.cpp

```diff
@@ -350,50 +377,85 @@ void prologSwizzle(TensorView* shared_mem_tv, const MatmulParams& params) {
      *   7|          |          |
      *    +----------+----------+
      */
+    // The first para in std::gcd is from the above derivation.
+    // The second para is from the fact that we need to split tile_size_y by
+    // n_cols and then by swizzle_period. If swizzle_period is smaller than the
+    // first para, then the bank conflict can't be fully removed.
+    int64_t swizzle_period =
+        std::gcd(n_rows / repeated_pattern_size, tile_size_y / n_cols);
+    // update repeated_pattern_size to ensure n_rows / repeated_pattern_size
+    // equals to swizzle_period, this is required by 2D swizzle.
+    repeated_pattern_size = n_rows / swizzle_period;
```

Comment:
With the `gcd` above, `swizzle_period` is either equal or smaller than `n_rows / repeated_pattern_size`, I don't think we should change `repeated_pattern_size`, because by definition for `i = 0...repeated_pattern_size`, different `i` values never conflict. Instead, we should check that, if `swizzle_period < n_rows / repeated_pattern_size`, we split `n_rows` as 3d `[outer, swizzle_period, repeated_pattern_size]` and swizzle the `swizzle_period` with matrix id. In the future, we might want to just merge the `tile_x/n_rows` with `tile_y/n_cols` and use that for swizzle, but let's leave it in future PR.

Reply:
Added the following code to split n_rows as 3d  :
```
    // further split n_rows to [outer, swizzle_period, repeated_pattern_size]
    if (swizzle_period < n_rows / repeated_pattern_size) {
      const int repeat_axis = repeated_pattern_size > 1 ? -4 - skip : -3 - skip;
      shared_mem_tv->split(repeat_axis, swizzle_period);
      //     -6      -5             -4              -3       -2         -1
      // [matrix id, nrows_outer, swizzle_period, pattern, matrix id, matrix,
      // ***** skip ***** ]
    }
```

Reply:
with this change, the bank conflict can be removed in cases previously can't be fully removed.